src/app/hooks/useRequest.js

```diff
@@ -3,11 +3,13 @@ import { useEffect } from 'react';
 import useLazyRequest from './useLazyRequest';
 
 const useRequest = ({ request, payload }) => {
-  const [isLoading, submitError, response, sendRequest] = useLazyRequest({ request, payload });
+  const [isLoading, submitError, response, sendRequest] = useLazyRequest({ request });
 
   useEffect(() => {
-    sendRequest(payload);
-  }, [payload, sendRequest]);
+    if (!isLoading && !response) {
```

Comment:
I don't understand this condition. Why should we check this? This implies that there are times where this hook is executing the request when it shouldn't so you set this condition, but I think it might be avoidable

Reply:
I added the useCallback hook in the next PR in order to test this problem with both get requests